packages/next/lib/verifyTypeScriptSetup.ts

```diff
@@ -28,7 +30,9 @@ export async function verifyTypeScriptSetup(
   imageImportsEnabled: boolean,
   cacheDir?: string
 ): Promise<{ result?: TypeCheckResult; version: string | null }> {
-  const tsConfigPath = path.join(dir, 'tsconfig.json')
+  const config = await loadConfig(PHASE_PRODUCTION_BUILD, dir)
+  const tsConfigName = config.typescript.tsconfigPath
+  const tsConfigPath = path.join(dir, tsConfigName!)
```

Comment:
You won't need `!` if you pass in the `NextConfigComplete` object

Reply:
TypeScript was complaining that `tsconfigPath` could be `undefined` since it's optional in `NextConfig` (and `NextConfigComplete`). Since there's a default, should I remove the `?` here: https://github.com/Dev-CasperTheGhost/next.js/blob/49da6328b14494ee9ad76359fadc42d303cd0d7d/packages/next/server/config-shared.ts#L38?

Reply:
It should probably be non-nullable on the Complete object here 
https://github.com/Dev-CasperTheGhost/next.js/blob/49da6328b14494ee9ad76359fadc42d303cd0d7d/packages/next/server/config-shared.ts#L9-L11

Reply:
Resolved now, Thanks!